### PR TITLE
Add online count to Lobby

### PR DESCRIPTION
**Pull request by GrantBartlett**
online-count

Could also possibly look at: 
- Changing the online count when the channel is changed.
- Making the Text customizable in an INI file somewhere.

### CODE REVIEW

File: DXMainClient/DXGUI/Multiplayer/CnCNet/CnCNetLobby.cs [hunk 1]
```diff
@@ -239,6 +247,10 @@ public override void Initialize()
             AddChild(lblCurrentChannel);
             AddChild(ddCurrentChannel);
             AddChild(playerContextMenu);
+            AddChild(lblOnline);
+
+            CnCNetPlayerCountTask.CnCNetGameCountUpdated += OnCnCNetGameCountUpdated;
+            CnCNetPlayerCountTask.ForceRefresh();
```
Comment (Rampastring) [inline DXMainClient/DXGUI/Multiplayer/CnCNet/CnCNetLobby.cs]: Why is ForceRefresh necessary?

File: DXMainClient/DXGUI/Multiplayer/CnCNet/CnCNetLobby.cs [hunk 2]
```diff
@@ -247,6 +259,14 @@ public override void Initialize()
             PostUIInit();
         }
 
+        private void OnCnCNetGameCountUpdated(object sender, PlayerCountEventArgs e)
+        {
+            lock (locker)
+            {
+                lblOnline.Text = "Online: " + e.PlayerCount.ToString();
+            }
```
Comment (Rampastring) [inline DXMainClient/DXGUI/Multiplayer/CnCNet/CnCNetLobby.cs]: Locking might not be necessary here, although MainMenu does it too. I'll check later if it's really needed.

File: DXMainClient/DXGUI/Multiplayer/CnCNet/CnCNetLobby.cs [hunk 3]
```diff
@@ -224,6 +226,12 @@ public override void Initialize()
             lblCurrentChannel.FontIndex = 1;
             lblCurrentChannel.Text = "CURRENT CHANNEL:";
 
+            lblOnline = new XNALabel(WindowManager);
+            lblOnline.Name = "lblOnline";
+            lblOnline.ClientRectangle = new Rectangle(310, 14, 0, 0);
+            lblOnline.FontIndex = 1;
+            lblOnline.Text = "";
```
Comment (Rampastring) [inline DXMainClient/DXGUI/Multiplayer/CnCNet/CnCNetLobby.cs]: Call lblOnline.Disable() here to make it disabled by default. Then you can enable it for YR in CnCNetLobby.ini.

**Comment by GrantBartlett** [comment]
- Removed locker
- Replaced ForceUpdater with a cached online count.
- Seperated lblOnline and lblOnlineCount into two separate labels so they can be edited fully.
- Lobby count disabled by default

**Comment by GrantBartlett** [comment]
@Rampastring ^

File: DXMainClient/Domain/Multiplayer/CnCNet/CnCNetPlayerCountTask.cs [hunk 4]
```diff
@@ -86,6 +82,8 @@ private static int GetCnCNetPlayerCount()
                 return -1;
             }
         }
+
+        public static int PlayerCount { get; set; }
```
Comment (Rampastring) [inline DXMainClient/Domain/Multiplayer/CnCNet/CnCNetPlayerCountTask.cs]: Move this to the beginning of the class (like the other fields) and make the setter private.

**Pull request merged**